src/audio/tdfb/tdfb.c

```diff
@@ -426,6 +651,16 @@ static int tdfb_copy(struct comp_dev *dev)
 		}
 	}
 
+	/* Handle enum controls */
+	if (cd->update) {
+		cd->update = false;
+		ret = tdfb_setup(cd, sourceb->stream.channels, sinkb->stream.channels);
```

Comment:
if setup failed do we fall back to the old values or enter a broken state?

Reply:
The return value can be -EINVAL or -ENOMEM. The pipeline should stop the copy() scheduring if a component returns error. I haven't tested it though. I'll check it happens.